demos/twitter/src/main/java/com/datatorrent/demos/twitter/RSSFeedExtractor.java

```diff
@@ -0,0 +1,160 @@
+import com.datatorrent.api.Context.OperatorContext;
+import com.datatorrent.api.InputOperator;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+/**
+ * <p>
+ * TwitterStatusUserExtractor class.
+ * </p>
+ *
+ * @since 0.3.2
+ */
+public class RSSFeedExtractor implements InputOperator
+{
+  protected String URLToConvert = "";
+  protected BufferedReader in;
+  URL rssURL;
+  public final transient DefaultOutputPort<String> output = new DefaultOutputPort<String>();
+
+  @Override
+  public void beginWindow(long windowId)
+  {
+    // TODO Auto-generated method stub
+  }
+
+  @Override
+  public void endWindow()
+  {
+    // TODO Auto-generated method stub
+
+  }
+
+  @Override
+  public void setup(OperatorContext context)
+  {
+    try {
+      DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
+      DocumentBuilder dBuilder = dbFactory.newDocumentBuilder();
+      InputStream is = this.getClass().getClassLoader()
+          .getResourceAsStream("RSS.xml");
+      Document doc = null;
+      try {
+        doc = dBuilder.parse(is);
+      } catch (SAXException e) {
+        throw new RuntimeException("SAX Exception");
+      }
+
+      doc.getDocumentElement().normalize();
+
+      NodeList nList = doc.getElementsByTagName("sub");
+
+      for (int index = 0; index < nList.getLength(); index++) {
+        Node nNode = nList.item(index);
+        if (nNode.getNodeType() == Node.ELEMENT_NODE) {
+          Element eElement = (Element) nNode;
+          URLToConvert = eElement.getElementsByTagName("value").item(0)
+              .getTextContent();
+        }
+      }
+
+    } catch (ParserConfigurationException e) {
+      throw new RuntimeException("Parser Configuration Exception");
+
+    } catch (IOException e) {
+      throw new RuntimeException("Runtime Exception");
+    }
+
+    try {
+      rssURL = new URL(URLToConvert);
+    } catch (MalformedURLException e) {
+      throw new RuntimeException("Malformed URL Exception");
+    }
+  }
+
+  @Override
+  public void teardown()
+  {
+    // TODO Auto-generated method stub
+    try {
+      in.close();
+    } catch (IOException e) {
+      throw new RuntimeException("IO Exception");
+    }
+  }
+
+  @Override
+  public void emitTuples()
+  {
+    // TODO Auto-generated method stub
+    try {
+      in = new BufferedReader(new InputStreamReader(rssURL.openStream()));
+      String sourceCode = "";
+      String line;
+      while ((line = in.readLine()) != null) {
+        if (line.contains("<title>")) {
```

Comment:
Can you use RegEx patttern capture for this? The Patterns library of Java is optimized for such capture operations.